cohttp-lwt/src/server.ml

```diff
@@ -109,16 +109,20 @@ module Make(IO:S.IO) = struct
       (fun () ->
          Lwt.catch
            (fun () -> callback conn req body)
-           (fun exn ->
-             Log.err (fun f -> f "Error handling %a: %s\n%!" Request.pp_hum req (Printexc.to_string exn));
-             respond_error ~body:"Internal Server Error" () >|= fun rsp ->
-             `Response rsp
+           (function
+             | Out_of_memory -> raise Out_of_memory
```

Comment:
As much as I agree with your rationale, what if we make this change optional with a new optional parameter?

Also why have you decided to use `raise` instead of `Lwt.fail`, it makes less allocations in such pressure situation?

Reply:
@mseri as mentioned above, I'd be ok with this behaviour being optional. Oh, I wasn't aware whether `raise` or `Lwt.fail` would be preferred. If it is the latter, that's fine with me as well.

Reply:
To be honest I don't know. `Lwt.catch` will catch both. I tend to use `Lwt.fail` only because wraps the failure in a `Lwt` promise and composes better.